openstates/de/legislators.py

```diff
@@ -68,64 +92,64 @@ def scrape_bio(self, term, chamber, district, name, url):
+        office_list = doc.xpath("//tr")
+        for office in office_list:
+            title_td = 0
+            #in some trs the photo is the first td
+            if len(office.xpath("./td/img")) > 0:
+                title_td = 1
+            try:
+                title_text = office.xpath("./td")[title_td].text_content().lower()
+                content = office.xpath("./td")[title_td+1].text_content()
+
+            except IndexError:
+                continue
+
+            if "legislative office" in title_text:
+                leg_office["address"] = content.strip()
+            if "legislative phone" in title_text:
+                phones = content.lower().split("\n")
+                if len(phones) == 1:
+                    phone = phones[0].replace("phone:","").strip()
+                    if phone:
+                        leg_office["phone"] = phone
+                else:
+                    for line in phones:
+                        if "phone" in line:
+                            leg_office["phone"] = line.replace("phone:","").strip()
+                        elif "fax" in line:
+                            leg_office["fax"] = line.replace("fax:","").strip()
+
+            if "outside office" in title_text:
+                dist_office["address"] = content.strip()
+            if "outside phone" in title_text:
+                phones = content.lower().split("\n")
+                if len(phones) == 1:
+                    phone = phones[0].replace("phone:","").strip()
+                    if phone:
+                        dist_office["phone"] = phone
                 else:
-                    office['address'] = chunks[0]
-                    numbers = []
-                    offices.append(office)
-                    continue
-            else:
-                if not chunks:
-                    # This office has no data.
-                    continue
-                address = chunks.pop(0)
-                numbers = chunks
-                office['address'] = address
-            for number in numbers:
-                for line in number.splitlines():
-                    if not line.strip():
-                        continue
-                    for key in ('phone', 'fax'):
-                        if key in line.lower():
-                            break
-                    number = re.search('\(\d{3}\) \d{3}\-\d{4}', line)
-                    if number:
-                        number = number.group()
-                        office[key] = number
-                offices.append(office)
-
-        return dict(info, offices=offices)
+                    for line in phones:
+                        if "phone" in line:
+                            dist_office["phone"] = line.replace("phone:","").strip()
+                        elif "fax" in line:
+                            dist_office["fax"] = line.replace("fax:","").strip()
+
+        offices = [o for o in [leg_office,dist_office] if o["address"]]
```

Comment:
Maybe `assert` that at least one office has an `address`? Otherwise, we may be throwing out useful email and phone data.